### PR TITLE
Simplify multi-lining binop exprs

### PR DESCRIPTION
**Pull request by nrc**
r? @scampi

### CODE REVIEW

File: src/pairs.rs [hunk 1]
```diff
@@ -83,33 +83,22 @@ fn rewrite_pairs_one_line<T: Rewrite>(
...
-        if let Some(rewrite) = rewrite_pairs_multiline(multiline_list, multiline_shape, context) {
-            // Also, don't let expressions surrounded by parens go multi-line,
-            // this looks really bad.
-            if rewrite.starts_with('(') || is_single_line(&rewrite) {
-                return None;
-            }
-        }
+    // Check the last expression in the list. We sometimes let this expression
+    // go over multiple lines, but we check for some ugly conditions.
+    if !(is_single_line(&result) || last_rewrite.starts_with('{'))
```
Comment (scampi) [inline src/pairs.rs]: do you have an example of that last part starting with `'{'` ?

**Comment by otavio** [comment]
Shouldn't it go to multi-line only when it does not fit on the available width?

If you look at the reformatted code, of `rustfmt` itself, it looks weird in cases all binop expressions could fit a single line.

**Comment by nrc** [comment]
> Shouldn't it go to multi-line only when it does not fit on the available width?

We have a heuristic max width for chains which most of these hit. It wouldn't multiline below that width.

File: src/pairs.rs (hunk 1 above)
Comment (nrc) [reply src/pairs.rs]: There were some places in rustfmt that had a block and so the opening line was just `+ {` or something. It looked weird.

**Pull request merged**